src/System.Net.Primitives/src/System/Net/CredentialCache.cs

```diff
@@ -167,6 +161,11 @@ public NetworkCredential GetCredential(Uri uriPrefix, string authenticationType)
                 GlobalLog.Print("CredentialCache::GetCredential(uriPrefix=\"" + uriPrefix + "\", authType=\"" + authenticationType + "\")");
             }
 
+            if (_cache == null)
+            {
+                return null;
+            }
+
```

Comment:
> This ends up changing the logging by short-circuiting the logging call at the end of the function.

The `Remove` methods already short-circuit without logging when any of the parameters are null, so I did the same in both `Remove` and `GetCredential` when the dictionary is null. Note that for the `GetCredential` overloads, it will log entry of the method before the short-circuit, but as @stephentoub points out, with the way it short-circuits currently, it won't log the result of the lookup.

> I recommend that logging calls be added to these places that 'short-circuit' the original code paths.

OK, I'll add new log messages about the dictionary being null before short-circuiting.